src/component/Filter.jsx

```diff
@@ -0,0 +1,251 @@
+import React from "react";
+import { useFilter, FILTER_ACTION, filterinitialstate } from "../state/index";
+
+const Filter = ({ displayMobileFilter }) => {
+  const {
+    filterDispatch,
+    filterstate: { sort_by, rating, price_range, category },
+  } = useFilter();
+  const categoryHandler = (e) => {
+    const categroySelected = e.target.value;
+    const contains = category.includes(categroySelected);
+    if (contains) {
+      filterDispatch({
+        type: FILTER_ACTION.SORT_BY_CATEGORY,
+        payload: {
+          ...filterinitialstate,
+          category: [...category.filter((a) => a !== categroySelected)],
+        },
+      });
+    } else {
+      filterDispatch({
+        type: FILTER_ACTION.SORT_BY_CATEGORY,
+        payload: {
+          ...filterinitialstate,
+          category: category.concat(categroySelected),
+        },
+      });
+    }
+  };
+
+  return (
+    <section
+      className={`filter_container ${
+        !displayMobileFilter ? "displayFilter" : ""
+      }`}
+    >
+      <div className="heading_clearall_btn">
+        <h1>Filters</h1>
+
+        <p
+          onClick={() =>
+            filterDispatch({
+              type: FILTER_ACTION.RESET_FILTER,
+              payload: { ...filterinitialstate },
+            })
+          }
+          className="btn btn-primary btn-link"
+        >
+          Clear All
+        </p>
+      </div>
+      <div className="price-slider">
+        <h1>Price</h1>
+        <div className="slider-container">
+          <input
+            type="range"
+            onChange={(e) => {
+              filterDispatch({
+                type: FILTER_ACTION.SORT_BY_PRICE,
+                payload: { ...filterinitialstate, price_range: e.target.value },
+              });
+            }}
+            value={price_range}
+            min="100"
+            max="1500"
+            className="slider"
+          />
+        </div>
+        <div className="price_range">
+          <h1>From:100</h1>
+          <h1>From:1500</h1>
+        </div>
+      </div>
+
+      <ul className="sim-unList">
+        <div className="list-heading">Filter on the basis of Rating</div>
```

Comment:
It is best practice to only have `<li>` inside `<ul>`. You can make the list-heading li

Reply:
i have made same in my cl .i will make changes in cl first